tests/test_gossip.py

```diff
@@ -180,3 +180,45 @@ def test_mute_accepts_only_lists(arg):
     with pytest.raises(TypeError):
         with gossip.mute_context(arg):
             pass
+
+
+def test_mute_group(checkpoint):
+
+    @gossip.register('a.b.c')
+    def handler():
+        checkpoint()
+
+    gossip.get_group("a").should_not_be_muted()
+
+    with pytest.raises(CannotMuteHooks):
+        with gossip.mute_context(['a.b.c']):
+            gossip.trigger('a.b.c')
+    assert not checkpoint.called
+
+
+def test_mute_hook(checkpoint):
+
+    @gossip.register('a.b.c')
+    def handler():
+        checkpoint()
+
+    hook = gossip.define('a.b.d')
+    hook.should_not_be_muted()
+    assert not hook.can_be_muted()
+
+    with gossip.mute_context(['a.b.c']):
+        gossip.trigger('a.b.c')
+    assert not checkpoint.called
+
+
+def mute_other_sub_group():
```

Comment:
more tests should be added for muting subgroups inside unmuted groups, hooks and vice-versa (whether we forbid it or not)